frontend/src/lib/components/Cards/TextCard/TextCard.tsx

```diff
@@ -0,0 +1,289 @@
+import ReactMarkdown from 'react-markdown'
+import { IconMarkdown } from 'lib/components/icons'
+import { Tabs } from 'antd'
+import { UserActivityIndicator } from 'lib/components/UserActivityIndicator/UserActivityIndicator'
+import { textCardModalLogic } from 'lib/components/Cards/TextCard/textCardModalLogic'
+import { dashboardsModel } from '~/models/dashboardsModel'
+
+export function TextTileModal({
+    isOpen,
+    onClose,
+    dashboard,
+    textTileId,
+}: {
+    isOpen: boolean
+    onClose: () => void
+    dashboard: DashboardType
+    textTileId: number | 'new' | null
+}): JSX.Element {
+    const modalLogic = textCardModalLogic({ dashboard, textTileId: textTileId ?? 'new', onClose })
+    const { isTextTileSubmitting } = useValues(modalLogic)
+    const { submitTextTile, resetTextTile } = useActions(modalLogic)
+
+    const handleClose = (): void => {
+        resetTextTile()
+        onClose()
+    }
+
+    return (
+        <LemonModal
+            closable={true}
+            isOpen={isOpen}
+            title={''}
+            onClose={handleClose}
+            footer={
+                <>
+                    <LemonButton disabled={isTextTileSubmitting} type="secondary" onClick={handleClose}>
+                        Cancel
+                    </LemonButton>
+                    <LemonButton
+                        disabled={isTextTileSubmitting}
+                        loading={isTextTileSubmitting}
+                        form="text-tile-form"
+                        htmlType="submit"
+                        type="primary"
+                        onClick={submitTextTile}
+                        data-attr={textTileId === 'new' ? 'save-new-text-tile' : 'edit-text-tile-text'}
+                    >
+                        Save
+                    </LemonButton>
+                </>
+            }
+        >
+            <Form
+                logic={textCardModalLogic}
+                props={{ dashboard: dashboard, textTileId: textTileId }}
+                formKey={'textTile'}
+                id="text-tile-form"
+                className=""
+                enableFormOnSubmit
+            >
+                <Field name="body" label="">
+                    {({ value, onChange }) => (
+                        <Tabs>
+                            <Tabs.TabPane tab="Write" key="write-card" destroyInactiveTabPane={true}>
+                                <LemonTextArea
+                                    data-attr="text-card-edit-area"
+                                    autoFocus
+                                    value={value}
+                                    onChange={(newValue) => onChange(newValue)}
+                                />
+                                <div className="text-muted inline-flex items-center space-x-1">
+                                    <IconMarkdown />
+                                    <span>Basic formatting and markdown support</span>
```

Comment:
Isn't Markdown a way of formatting text? This is kind of like "Basic coding and Python" support – these things seem to be on different levels 😅 
```suggestion
                                    <span>Markdown formatting support</span>
```